testdata/project-v4/.golangci.yml

```diff
@@ -36,13 +18,44 @@ linters:
     - prealloc
     - revive
     - staticcheck
-    - typecheck
     - unconvert
     - unparam
     - unused
-
-linters-settings:
-  revive:
+  settings:
+    revive:
+      rules:
+        - name: comment-spacings
+        - name: import-shadowing
+  exclusions:
+    generated: lax
     rules:
-      - name: comment-spacings
-      - name: import-shadowing
+      - linters:
+          - lll
+        path: api/*
+      - linters:
+          - dupl
+          - lll
+        path: internal/*
+      - linters:
+          - staticcheck
+        text: "ST1019"
+      - linters:
+          - staticcheck
+        text: "ST1005"
+      - linters:
+          - staticcheck
+        text: "QF1008"
```

Comment:
Why are we adding here?
Can we not specific ignore thing with // nolint: comment instead ?
could we ensure that the config after the upgrade is exactly == before the upgrade?

Reply:
Just for `inline` parameters. like: `ObjectMeta` and `GVK`